test/parallel_api/sycl_canary.pass.cpp

```diff
@@ -0,0 +1,73 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+// This test is a simple standalone SYCL test which is meant to prove that the SYCL installation is correct.
+// If this test fails, it means that the SYCL environment has not be configured properly.
+
+#if ((defined(CL_SYCL_LANGUAGE_VERSION) || defined(SYCL_LANGUAGE_VERSION)) &&                                         \
+     (__has_include(<sycl/sycl.hpp>) || __has_include(<CL/sycl.hpp>))) &&                                             \
+    (!defined(ONEDPL_USE_DPCPP_BACKEND) || ONEDPL_USE_DPCPP_BACKEND != 0)
+#define TEST_DPCPP_BACKEND_PRESENT 1
+#else
+#define TEST_DPCPP_BACKEND_PRESENT 0
+#endif
+
+#if TEST_DPCPP_BACKEND_PRESENT
+
+#if __has_include(<sycl/sycl.hpp>)
+#    include <sycl/sycl.hpp>
+#else
+#    include <CL/sycl.hpp>
+#endif
+
+
+// Combine SYCL runtime library version
+#if defined(__LIBSYCL_MAJOR_VERSION) && defined(__LIBSYCL_MINOR_VERSION) && defined(__LIBSYCL_PATCH_VERSION)
+#    define TEST_LIBSYCL_VERSION                                                                                    \
+        (__LIBSYCL_MAJOR_VERSION * 10000 + __LIBSYCL_MINOR_VERSION * 100 + __LIBSYCL_PATCH_VERSION)
+#else
+#    define TEST_LIBSYCL_VERSION 0
+#endif
+
+
+inline auto default_selector =
+#    if TEST_LIBSYCL_VERSION >= 60000
+        sycl::default_selector_v;
+#    else
+        sycl::default_selector{};
+#    endif
+
+class canary_test_name;
+
+void
+test()
+{
+    sycl::queue q(default_selector);
+    {
+        q.submit([&](sycl::handler& cgh) {
+            cgh.single_task<canary_test_name>([=]() {});
+        });
+    }
+}
+#endif // TEST_DPCPP_BACKEND_PRESENT
+
+int
+main()
+{
+#if TEST_DPCPP_BACKEND_PRESENT
+    test();
+#endif
+    return 0;
```

Comment:
I want to divorce it from the oneDPL tests infrastructure as much as possible so that it is clear when there is a failure, it is not coming from oneDPL's code changes.  We probably can work out a way for it to be a skipped test from the internal CI though so it doesn't impact the passrate.

Reply:
I see. From the CTest side in CI, I think we can just exclude it then when running the actual CI after the canary test with the `-E` flag.

Reply:
I wanted to leave `-E` available for users.  It looks like if you give cmake multiple `-E` filters it only uses the last one.